Detectors/TPC/simulation/include/TPCSimulation/DigitContainer.h

```diff
@@ -61,24 +59,23 @@ class DigitContainer{
 
     /// Fill output TClonesArray
     /// \param output Output container
-    void fillOutputContainer(TClonesArray *output, int eventTime=0, bool isContinuous=true);
-
-    /// Fill output TClonesArray
-    /// \param output Output container
-    void fillOutputContainer(TClonesArray *output, std::vector<CommonMode> &commonModeContainer);
-
-    /// Process Common Mode Information and sum up all charges in a given CRU and time bin
-    /// \param output Output container
-    void processCommonMode(std::vector<CommonMode> &);
+    /// \param debug Optional debug output container
+    /// \param eventTime time stamp of the event
+    /// \param isContinuous Switch for continuous readout
+    void fillOutputContainer(TClonesArray *output, TClonesArray *debug, int eventTime=0, bool isContinuous=true);
 
   private:
     std::array<std::unique_ptr<DigitCRU> , CRU::MaxCRU> mCRU;   ///< CRU Container for the ADC value
+    std::unique_ptr<CommonModeContainer>                mCommonModeContainer; ///< Container for the common mode values
 };
 
 inline
 DigitContainer::DigitContainer()
-  : mCRU()
-{}
+  : mCRU(),
+    mCommonModeContainer()
+{
+  mCommonModeContainer = std::make_unique<CommonModeContainer>();
```

Comment:
why don't you put the make_unique in the initializer list?
`mCommonModeContainer(std::make_unique<CommonModeContainer>())`

By the way, do we need a pointer type here? It seems that a common mode container will always be created, so it could also be a simple instance in the header:
`CommonModeContainer mCommonMondeContainer;`